Automattic/wp-calypso
# Stats: use Redux for Post Trends

Part of moving stats away from the `stats-list` and `localStore` - this branch moves the Post Trends visualization on the insights page to use redux-backed data.  In order to do so, a new data component `QuerySiteStats` has been added here, which will be utilized throughout the rest of the stats components to fetch data from the API.

Alongside the addition of `QuerySiteStats`, I realized the initial approach taken in #5743 WRT the response structure of the stats streak data was incorrect.  The updates to the selectors and tests to fix this can be seen in 7c11823.

Integrating the new query component, and associated selector logic can be seen in b668a54.

__To Test__
- Open up the stats insights page, validate the "Posting Activity" component up top looks/functions as it does in production
- Switch sites to verify loading state and data updates between site changes

There is quite a bit more ES6-ification that can happen throughout the `post-trends/*.jsx` files, but I tried to keep this focused on the task at hand.

Test live: https://calypso.live/?branch=update/stats/insights-streak
